### PR TITLE
Add KMS for Apple keychain and Secure Enclave

### PR DESCRIPTION
**Pull request by maraino**
This commit adds the initial version of mackms, a kms for Apple keychain, and Secure Enclave. This version supports the KeyManager interface and an extra method for deleting keys in the Secure Enclave.

### CODE REVIEW

File: kms/mackms/mackms_test.go [hunk 1]
```diff
@@ -0,0 +1,705 @@
...
+	"go.step.sm/crypto/randutil"
+)
+
+func createPrivateKeyOnly(t *testing.T, name string, signatureAlgorithm apiv1.SignatureAlgorithm) (*apiv1.CreateKeyResponse, error) {
+	t.Helper()
+
+	u, err := parseURI(name)
+	if err != nil {
+		return nil, err
+	}
```
Comment (hslatman) [inline kms/mackms/mackms_test.go:49]: This instance and the others can be replaced with `require.NoError(t, err)` and not return the error.

File: kms/mackms/mackms_test.go [hunk 2]
```diff
@@ -0,0 +1,705 @@
+//go:build darwin && cgo && !nomackms
```
Comment (hslatman) [inline kms/mackms/mackms_test.go:1]: This probably needs the CI to be adapted to also run on macOS

File: kms/mackms/mackms_test.go (hunk 1 above)
Comment (maraino) [reply kms/mackms/mackms_test.go:49]: I wasn't sure if the defer functions were called, but I tested, and it does, `t.FailNow()` calls to `runtime.Goexit()`, which runs all deferred calls.

File: kms/mackms/mackms_test.go (hunk 2 above)
Comment (maraino) [reply kms/mackms/mackms_test.go:1]: This is something I considered doing but I decided to leave it for a future PR.

**Comment by maraino** [comment]
@hslatman I've fixed most of your comments with 250ebe0, and change the MacKMS comment with d7a8c97

**Review by hslatman — APPROVED** [review]

**Review by dopey — APPROVED** [review]

**Pull request merged**